components/FormularAddRecipe.js

```diff
@@ -0,0 +1,197 @@
+import styled from "styled-components";
+import { useState } from "react";
+import { nanoid } from "nanoid";
+
+const StyledForm = styled.form`
+  display: flex;
+  align-items: left;
+  gap: 0.5 rem;
+  flex-direction: column;
+  border: solid black, 1em;
+  padding: 1 em;
+`;
+
+const StyledDiv = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 500px;
+  border: 1px solid pink;
+`;
+
+const StyledInput = styled.div`
+  background-color: lightgrey;
+  margin: 10px;
+  padding: 10px;
+  border-radius: 5px;
+`;
+
+const StyledIngredientsSection = styled.section`
+  background-color: lightgrey;
+  margin: 10px;
+  padding: 10px;
+  border-radius: 5px;
+  display: flex;
+  gap: 5px;
+`;
+
+const StyledIngredientInput = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 150px;
+`;
+
+const StyledButton = styled.button`
+  background-color: hotpink;
+  width: 480px;
+  margin: 10px;
+  height: 40px;
+  font-size: 18px;
+  border-radius: 5px;
+  align-item: center;
+`;
+
+export default function RecipeForm({ onSubmit, formName, defaultData }) {
+  const [ingredients, setIngredients] = useState([]);
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+    onSubmit({ ...data, ingredients });
+  }
+
+  return (
+    <StyledDiv>
+      <StyledForm
+        aria-labelledby={formName}
+        id="recipeForm"
+        onSubmit={handleSubmit}
+      >
+        <StyledInput>
+          <label htmlFor="title">Title</label>
+          <input
+            id="title"
+            name="title"
+            type="text"
+            defaultValue={defaultData?.title}
+          />
+        </StyledInput>
+        <StyledInput>
+          <label htmlFor="preparation">
+            preparation:
+            <select name="preparation" defaultValue="Oven">
```

Comment:
this select is missing its id-attribute to connect it with the label:

```suggestion
            <select name="preparation" defaultValue="Oven" id="preparation">
```